nibabel/streamlines/tests/test_streamlines.py

```diff
@@ -0,0 +1,256 @@
+    def test_load_simple_file(self):
+        for lazy_load in [False, True]:
+            for simple_filename in self.simple_filenames:
+                tfile = nib.streamlines.load(simple_filename,
+                                             lazy_load=lazy_load)
+                assert_true(isinstance(tfile, TractogramFile))
+
+                if lazy_load:
+                    assert_true(type(tfile.tractogram), Tractogram)
+                else:
+                    assert_true(type(tfile.tractogram), LazyTractogram)
+
+                assert_tractogram_equal(tfile.tractogram,
+                                        self.simple_tractogram)
+
+    def test_load_complex_file(self):
+        for lazy_load in [False, True]:
+            for complex_filename in self.complex_filenames:
+                tfile = nib.streamlines.load(complex_filename,
+                                             lazy_load=lazy_load)
+                assert_true(isinstance(tfile, TractogramFile))
+
+                if lazy_load:
+                    assert_true(type(tfile.tractogram), Tractogram)
+                else:
+                    assert_true(type(tfile.tractogram), LazyTractogram)
+
+                tractogram = Tractogram(self.streamlines)
+
+                if tfile.support_data_per_point():
+                    tractogram.data_per_point = self.data_per_point
+
+                if tfile.support_data_per_streamline():
+                    tractogram.data_per_streamline = self.data_per_streamline
+
+                assert_tractogram_equal(tfile.tractogram,
+                                        tractogram)
+
+    def test_save_empty_file(self):
+        tractogram = Tractogram()
+        for ext, cls in nib.streamlines.FORMATS.items():
+            with tempfile.NamedTemporaryFile(mode="w+b", suffix=ext) as f:
+                nib.streamlines.save_tractogram(tractogram, f.name)
+                tfile = nib.streamlines.load(f, lazy_load=False)
+                assert_tractogram_equal(tfile.tractogram, tractogram)
+
+    def test_save_simple_file(self):
+        tractogram = Tractogram(self.streamlines)
+        for ext, cls in nib.streamlines.FORMATS.items():
+            with tempfile.NamedTemporaryFile(mode="w+b", suffix=ext) as f:
+                nib.streamlines.save_tractogram(tractogram, f.name)
+                tfile = nib.streamlines.load(f, lazy_load=False)
+                assert_tractogram_equal(tfile.tractogram, tractogram)
+
+    def test_save_complex_file(self):
+        complex_tractogram = Tractogram(self.streamlines,
+                                        self.data_per_streamline,
+                                        self.data_per_point)
+
+        for ext, cls in nib.streamlines.FORMATS.items():
+            with tempfile.NamedTemporaryFile(mode="w+b", suffix=ext) as f:
+                with clear_and_catch_warnings(record=True, modules=[trk]) as w:
+                    nib.streamlines.save_tractogram(complex_tractogram, f.name)
```

Comment:
I think the problem here (and with the other permission errors) is that Windows doesn't like you re-opening the NamedTemporaryFile.

What I tend to do is to use the `nibabel.tmpdirs.InTemporaryDirectory` context manager and then load / save with some filename.   When you do this, you have to make sure all objects holding onto the file get deleted though, otherwise Windows (again) may refuse to delete the temporary file...